src/core/reducers/addons.js

```diff
@@ -385,31 +385,45 @@ export default function addonsReducer(
       const bySlug = { ...state.bySlug };
       const loadingBySlug = { ...state.loadingBySlug };
 
-      loadedAddons.forEach((loadedAddon) => {
+      return loadedAddons.reduce((acc, loadedAddon) => {
         const addon = createInternalAddon(loadedAddon);
-        // Flow wants hash maps with string keys.
-        // See: https://zhenyong.github.io/flowtype/docs/objects.html#objects-as-maps
-        byID[`${addon.id}`] = addon;
-
-        if (addon.slug) {
-          bySlug[addon.slug.toLowerCase()] = addon.id;
-          loadingBySlug[addon.slug.toLowerCase()] = false;
-        }
-
-        if (addon.guid) {
-          // `guid` is already "normalized" with the `getGuid()` function in
-          // `createInternalAddon()`.
-          byGUID[addon.guid] = addon.id;
-        }
-      });
-
-      return {
+        // $FLOW_FIXME
```

Comment:
@kumar303 I am not sure how to keep Flow happy with the usage of the object spread operator over strings - hence `$FLOW_FIXME`. How can we fix this?